### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by ianstormtaylor**
Hey @manast!

I wanted to help contribute a bit, so I took a stab at refactoring the readme and docs a bit to make it easier to find information, and to make it feel a bit better. I tried to fix any grammar issues I saw, keep the code samples (and Markdown formatting) consistent everywhere, and split things up a bit in easier to digest pieces. There's probably still more that can be done though!

(There are a bunch of commits, but I think you can use the "Squash & Merge" GitHub functionality to combine them into one. If not, let me know!)

I also added a very similar, but slightly nicer (hopefully) logo!

You can check out the [real-life update on my branch](https://github.com/ianstormtaylor/bull/tree/update-docs), and here's a screenshot...

Let me know what you think!

image

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+
+// Will listen globally, to instances of this queue...
+queue.on('global:completed', listener);
+```
+
+---
+
+
+Priorty Queue
+-------------
```
Comment (manast) [inline REFERENCE.md]: Priority

**Comment by manast** [comment]
Looks great man! Thanks for the effort.

**Comment by bradvogel** [comment]
Awesome!

**Comment by chuym** [comment]
💯

**Comment by wearhere** [comment]
This is so cool! Logo and everything

**Pull request merged**